File: app/src/main/java/com/sami/rippel/model/MyService.java
package com.sami.rippel.model;

import android.content.Context;

import com.jakewharton.retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import com.sami.rippel.model.api.service.WallpaperApiService;

import retrofit2.Retrofit;
import retrofit2.converter.simplexml.SimpleXmlConverterFactory;

/**
 * Created by yassin baccour on 10/04/2017.
 */

public class MyService {
    public Context context;
    public Retrofit retrofit;
    public RetrofitHelper mRetrofitHelper;

    public MyService() {
        mRetrofitHelper = new RetrofitHelper(getWallpaperApiService());
    }

    public WallpaperApiService getWallpaperApiService() {
        return createRetrofitXMlConverter().create(WallpaperApiService.class);
    }

    private Retrofit createRetrofitXMlConverter() {
        return new Retrofit.Builder()
                .baseUrl(Constants.BASE_URL)
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())  //Pour info j'utilise l'adapter de jakewharton
                .addConverterFactory(SimpleXmlConverterFactory.create()).build();
    }
}

File: app/src/main/java/com/sami/rippel/model/MyService.kt
package com.sami.rippel.model

import android.content.Context
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.simplexml.SimpleXmlConverterFactory

/**
 * Created by yassin baccour on 10/04/2017.
 */
class MyService {
    var context: Context? = null

    @JvmField
    var mRetrofitHelper: RetrofitHelper
    private val wallpaperApiService: WallpaperApiService
        get() = createRetrofitXMlConverter().create(WallpaperApiService::class.java)

    init {
        mRetrofitHelper = RetrofitHelper(wallpaperApiService)
    }

    private fun createRetrofitXMlConverter(): Retrofit {
        return Retrofit.Builder()
            .baseUrl(Constants.BASE_URL)
            .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
            .addConverterFactory(SimpleXmlConverterFactory.create()).build()
    }

}